testdata/p4_16_errors_outputs/loop3-err.p4-stderr

```diff
@@ -0,0 +1,4 @@
+loop3-err.p4(51):syntax error, unexpected IN
```

Comment:
As Chris pointed out it might be better to throw the error in the front end instead of the parser because this message is confusing and doesn't tell you at all what is going on.

Reply:
Agreed it would be nice to have a better error message.  My goal with this PR is simply to ensure we have at least one test case that the front end catches this error, with any error message at all.

Reply:
I think it would make sense to improve the parser errors in general. This is not just a matter of missing type if loops, the parser errors are quite confusing and hard to read in general. I'm not sure how much does the parser generator allow for customization of them though...

Reply:
No disagreement here -- everyone is free to volunteer for anything they want to work on here. :-).  A few people are fortunate enough to be paid to work on this project, but they might have other areas they want to focus on other than that, at this time.

Reply:
Just to be crystal clear in my intentions with this PR, if someone else wants to enhance the error message for this test program, be my guest.  I have no intention to do so myself.  If that means this PR sits and waits until that enhancement is made, that is fine with me.

Reply:
#4808 was merged a little prematurely, the intention here is to at least not forget about it. We can create an issue to keep track of this.